### PR TITLE
proxy API compatibility details update

### LINKED ISSUE #27512: Update mentions of removed proxy API
### MDN URL

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#proxy_api

### What specific section or headline is this issue about?

Proxy API section, most of it

### What information was incorrect, unhelpful, or incomplete?

The current content is as follows:

> Firefox's [proxy](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy) API followed a completely different design from Chrome's Proxy API.
>
> * In Firefox: An extension can register a PAC file.
> * In Chrome: An extension can register a PAC file, but can also define explicit proxying rules.
>
> Because this API is incompatible with Chrome's proxy API, the Firefox `proxy` API is only available through the `browser` namespace.

The only accurate part of this is that Firefox and Chrome's API differ.

Notably:
- Proxy extensions in Firefox cannot only register a PAC script. While they can do so through a different API (`autoConfigUrl` key in [`proxy.settings` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/settings), extensions can also use `proxy.onRequest` to provide ProxyInfo dynamically.
- It's not clear what "Chrome (...) can also define explicit proxying rules" is referring to. There is a way to declare the default proxy rules (https://developer.chrome.com/docs/extensions/reference/proxy/#type-ProxyRules), but that functionality is also available in Firefox through [`proxy.settings` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/settings).
- The `proxy` API is not only available in the `browser` namespace. It is also available in the `chrome` namespace, like any other extension API.

### What did you expect to see?

The documentation should be correct:

- It should continue to emphasize that the two APIs are completely different.
- Link to the two different API docs (Firefox's proxy API on MDN, Chrome's API at developer.chrome.com)
- Nice to have: compare the overlapping parts of the API.

### Do you have any supporting links, references, or citations?

- https://developer.chrome.com/docs/extensions/reference/proxy/
- https://bugzilla.mozilla.org/show_bug.cgi?id=1545811
- https://hg.mozilla.org/mozilla-central/rev/d5ccb3bf4c1b#l3.222 (source code showing that the special APIs in the defunct Proxy Scripts were only available through the "browser" namespace)

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Fixes  #27512

### CODE REVIEW

**Comment by rebloor** [comment]
@Rob--W did your request cover removing [proxy.register()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/register) and [proxy.unregister()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/unregister)?

File: files/en-us/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md [hunk 1]
```diff
@@ -107,12 +107,14 @@ If you call `notifications.create()` more than once in rapid succession:
...
 
-Firefox's {{WebExtAPIRef("proxy")}} API followed a completely different design from Chrome's Proxy API.
-
-- **In Firefox**: An extension can register a PAC file.
-- **In Chrome**: An extension can register a PAC file, but can also define explicit proxying rules.
-
-Because this API is incompatible with Chrome's `proxy` API, the Firefox proxy API is only available through the `browser` namespace.
+Firefox and Chrome include a Proxy API. However, the design of these two APIs is incompatible.
+
+- **In Firefox**: An extension cannot register a PAC file.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md:112]: This is incorrect. As noted in the issue, both Firefox and Chrome can register PAC files, but in different ways. That's what I meant when I wrote "The only accurate part of this is that Firefox and Chrome's API differ."

**Review by Rob--W — CHANGES_REQUESTED** [review]

**Comment by Rob--W** [comment]
> @Rob--W did your request cover removing [proxy.register()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/register) and [proxy.unregister()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/unregister)?

Not specifically. But given their age and irrelevance, I wouldn't mind them being removed.

**Comment by rebloor** [comment]
@Rob--W feedback addressed and proxy.register and proxy.register removed. Also see BCD update in Remove anachronistic proxy details ([#20513](https://github.com/mdn/browser-compat-data/pull/20513))

**Review by Rob--W — APPROVED** [review]

**Pull request merged**